### ISSUE TITLE
Can’t install Ruby on Mojave with MacPorts

### ISSUE THREAD

**Issue by FranklinYu** [postmodern/ruby-install#337]
The `mkmf.log` for RubyGems is reporting that it cannot find `netinet6/in6.h`. I guess that previously it’s finding it in `/usr/include`, but as we all know Apple removed that path since Mojave. On my machine the current path is:

```
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include/
```

The C compiler seems fine if I’m using it stand-alone. I tried compiling following snippet:

```c
#include <netinet/in.h>
int main() {}
int t(struct in6_addr *addr) {
    return IN6_IS_ADDR_UNSPECIFIED(addr);
}
```

And it works, although I have no idea which header it included. How to solve this and build Ruby on my machine?

### Note 1

I tried both compiling it with `cc test.c`, and with the actual command in `mkmf.log`:

```
clang -I../../.ext/include/x86_64-darwin18 -I../.././include -I../.././ext/socket -I../.. -I../../. -I/opt/local/include -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT -Wall -Wextra -Wpedantic -Wshadow -pipe -D__APPLE_USE_RFC_3542 -Werror
```

Both works without any warning/error.

### Note 2

I found that this only happens to Ruby 2.6, not Ruby 2.5 or below.

**Comment by havenwood** [comment]
It'd be awesome if you could [create an issue upstream](https://bugs.ruby-lang.org/projects/ruby-trunk/issues/new) on the CRuby bug tracker.

**Comment by FranklinYu** [comment]
@havenwood I’m not familiar with the installation process; do you think this issue should be report to the CRuby or [RubyGems](https://github.com/rubygems/rubygems/issues)?

**Comment by havenwood** [comment]
I was thinking CRuby's bugtracker, but good point - you could do either.

**Comment by FranklinYu** [comment]
I checked the log again, and found that it’s Ruby issue. The last error message is RubyGems complaining that it cannot `require 'socket'`, but that is caused by `socket` core extension failing to compile.

I created [bug 15665](https://bugs.ruby-lang.org/issues/15665).

**Comment by FranklinYu** [comment]
The bug has been fixed upstream, and will be available in 2.7 version. For 2.6 branch, user can manually apply the patch ruby/ruby@fe3ff5afb07e171fd950623c69abfbabbb2762a3 like

```
ruby-install -p https://github.com/ruby/ruby/commit/fe3ff5afb07e171fd950623c69abfbabbb2762a3.patch ruby-2.6.3
```

I’m still working with upstream to see whether they would backport this to 2.6 branch to avoid manual patch. Will report here for any update.
